src/ott/experimental/mmsinkhorn.py

```diff
@@ -0,0 +1,407 @@
+      iteration but every ``inner_iterations`` instead.
+    min_iterations: the minimum number of Sinkhorn iterations carried
+      out before the error is computed and monitored.
+    max_iterations: the maximum number of Sinkhorn iterations. If
+      ``max_iterations`` is equal to ``min_iterations``, Sinkhorn iterations are
+      run by default using a :func:`jax.lax.scan` loop rather than a custom,
+      unroll-able :func:`jax.lax.while_loop` that monitors convergence.
+      In that case the error is not monitored and the ``converged``
+      flag will return ``False`` as a consequence.
+    use_danskin: when ``True``, it is assumed the entropy regularized cost
+      is evaluated using optimal potentials that are frozen, i.e. whose
+      gradients have been stopped. This is useful when carrying out first order
+      differentiation, and is only valid mathematically when the algorithm has
+      converged with a low tolerance.
+    initializer: how to compute the initial potentials/scalings. This refers to
+      a few possible classes implemented following the template in
+      :class:`~ott.initializers.linear.SinkhornInitializer`.
+    progress_fn: callback function which gets called during the Sinkhorn
+      iterations, so the user can display the error at each iteration,
+      e.g., using a progress bar. See :func:`~ott.utils.default_progress_fn`
+      for a basic implementation.
+    kwargs_init: keyword arguments when creating the initializer.
+  """
+
+  def __init__(
+      self,
+      threshold: float = 1e-3,
+      norm_error: float = 1.0,
+      inner_iterations: int = 10,
+      min_iterations: int = 0,
+      max_iterations: int = 2000,
+      use_danskin: bool = True,
+  ):
+    self.threshold = threshold
+    self.inner_iterations = inner_iterations
+    self.min_iterations = min_iterations
+    self.max_iterations = max_iterations
+    self.norm_error = norm_error
+    self.use_danskin = use_danskin
+
+  def __call__(
+      self,
+      x_s: Tuple[jnp.ndarray, ...],
+      a_s: Optional[Tuple[jnp.ndarray, ...]] = None,
+      cost_fns: Optional[Union[costs.CostFn, Tuple[costs.CostFn, ...]]] = None,
+      epsilon: Optional[float] = None
+  ):
+    r"""Solves multimarginal OT for :math:`k` :math:`d`-dim point clouds.
+
+    Takes :math:`k` weights :math:`d`-dim point clouds and computes their
+    multimarginal optimal transport tensor
+
+    Args:
+      x_s: Tuple of :math:`k` point clouds, ``x_s[i]`` is a matrix of size
+      :math:`n_i\times d` where :math:`d` is a dimension common to all
+      point clouds.
+      a_s: Tuple of :math:`k` probability vectors, each of size :math:`n_i`.
+      cost_fns: Instance of :class:`~ott.solvers.geometry.costs.CostFn`, or
+        Tuple of :math:`k(k-1)/2` such instances. Note that the solver currently
+        assumes that these cost functions are symmetric. The cost function at
+        index :math:`i(k-\tfrac{i+1}{2})+j-i-1` will be used to compare
```

Comment:
i think this is possible, but I'd rather have a convenience wrapper for this. Since 99% of all use cases I envision will only involve one cost function, we would end up having `Union[costs.CostFn, Dict(...)` which would be a bit ugly.

For me the next evolution should be a generalized cost function taking a `k`-tuple of vectors.